File: Factorial algorithms/factorial.js
 // first variant via loop
        function factorial(n){
            let result = 1;
            for(let i = 1; i < n + 1; ++i){
                result *= i;
            }
            return result;    
        }

// Second variant  via recursion and ternary operator;
        function factorial(n){
            return n < 2 ? 1 : n * factorial(n - 1);
        }

// Third varsion via recusion 
        function factorial(n){
            if(n < 2) return 1;

            return n * factorial(n - 1);
        }

File: Factorial algorithms/factorial.java
static int factorial(int n) {
    return n > 1 ? n * factorial(n - 1) : 1;
}

// OR

static int factorial(int n) {
    int result = 1;
    for (int i = 2; i <= n; ++i) {
      result *= i;
    }

    return result;
}

// OR

static int factorial(int n) {
    if (n < 2)
      return 1;
    return n * factorial(n - 1);
}
